feat: allow WETH when adding liquidity to an existing LP position

Jdecristi opened the pull request:
<!-- Your PR title must follow conventional commits: https://github.com/Uniswap/interface#pr-title -->

fixes: #3705 

## Description
Allow users to add liquidity using WETH when adding liquidity to an existsing LP position

<!-- Delete this section if your change does not affect UI. -->
## Screen capture
<!--
### Before
|    Mobile    |   Desktop    |
| ------------ | ------------ |
| <img width="387" alt="Screenshot 2023-09-08 at 12 23 38 AM" src="https://github.com/Uniswap/interface/assets/89173284/4502d766-31dd-4313-b018-09cbc7138a70">|<img width="639" alt="Screenshot 2023-09-08 at 12 21 59 AM" src="https://github.com/Uniswap/interface/assets/89173284/6680731b-4c81-4330-a9f0-8e3290946bfd"> |


### After
|    Mobile    |   Desktop   |
| ------------ | ----------- |
|<img width="387" alt="Screenshot 2023-09-08 at 12 23 33 AM" src="https://github.com/Uniswap/interface/assets/89173284/a715f575-abf0-4109-a58c-407a18a61616">|<img width="639" alt="Screenshot 2023-09-08 at 12 22 05 AM" src="https://github.com/Uniswap/interface/assets/89173284/06c9e023-0c4a-4030-8efe-437a48d16084">|
|<img width="387" alt="Screenshot 2023-09-08 at 12 23 45 AM" src="https://github.com/Uniswap/interface/assets/89173284/aa516e88-8c2a-4e04-9803-d8ac1fc7efdf">|<img width="639" alt="Screenshot 2023-09-08 at 12 22 13 AM" src="https://github.com/Uniswap/interface/assets/89173284/2a346411-8a79-45ae-b5d2-08219bff8b2f">|
-->


https://github.com/Uniswap/interface/assets/89173284/eb60c35f-5502-4cde-b9c1-9da6cd8cf44e




## Test plan

<!-- Delete this section if your change is not a bug fix. -->
### Reproducing the error

<!-- Include steps to reproduce the bug. -->
1. navigate to [app.uniswap.org/pools](https://app.uniswap.org/#/pools)
2. Navigate to a pool using ETH or other native token
3. Make sure the toggle works and can still increase liquidity with WETH

just-toby commented on src/pages/AddLiquidity/UseWrappedCurrencyToggle.tsx:
```diff
@@ -0,0 +1,51 @@
+  onChangeCurrencies: (CurrencyAId?: string, currencyBId?: string) => void
+}
+
+export function UseWrappedCurrencyToggle({ currencyIdA, currencyIdB, onChangeCurrencies }: Props) {
```
```suggestion
export function WrappedCurrencyToggle({ currencyIdA, currencyIdB, onChangeCurrencies }: Props) {
```

just-toby commented on src/pages/AddLiquidity/UseWrappedCurrencyToggle.tsx:
```diff
@@ -0,0 +1,51 @@
+  return chainId !== undefined && currencyId === WRAPPED_NATIVE_CURRENCY[chainId]?.address
+}
+
+type Props = {
```
```suggestion
type WrappedCurrencyToggleProps = {
```

just-toby commented on src/pages/AddLiquidity/index.tsx:
```diff
@@ -873,9 +881,17 @@ function AddLiquidity() {
+                        {hasExistingPosition ? <Trans>Add more liquidity</Trans> : <Trans>Deposit Amounts</Trans>}
+                      </ThemedText.DeprecatedLabel>
+
+                      <UseWrappedCurrencyToggle
```
should we show this toggle if neither token is native or wrapped native?

infredible replied:
i don't think so. i've the user can't take action on it, it should be hidden

just-toby reviewed (changes requested):
@Jdecristi  can you upload a screen recording where you actually add liquidity using this new toggle? also, please rebase and fix the merge conflict

infredible commented:
Could you please make the toggle styling consistent with the one the "Selected range" row above? It'll reduce visual clutter and should help with overflow

![image](https://github.com/Uniswap/interface/assets/3887562/2cf7350b-e30b-4c25-90d4-bbd4d2a22286)

JFrankfurt commented:
I hate this bug and I'm so excited to have someone fixing it. tysm

Jdecristi commented:
@just-toby @JFrankfurt @infredible Sorry for the late response. Just started a new job, and things are hectic. I fixed the issues you brought up.

1. Code standards
2. Styling to match other toggles
3. Rebased and conflicts resloved

The video of me actually adding liquidity is in the Screen Capture section at the top

Cheers!

JFrankfurt commented:
heads up @Jdecristi the only thing stopping this from merging is the lint issue

JFrankfurt commented:
https://github.com/Jdecristi/uniswap-interface/pull/1

Jdecristi commented:
@JFrankfurt fixed!

JFrankfurt commented:
looks like you've got a failing unit test:

```
FAIL src/components/AccountDrawer/MiniPortfolio/Activity/utils.test.ts
  ● createGroups › should sort and group activities based on status and time

    expect(received).toContainEqual(expected) // deep equality

    Expected value: ObjectContaining {"title": "Today", "transactions": ArrayContaining [ObjectContaining {"status": "CONFIRMED", "timestamp": Any<Number>}]}
    Received array: [{"title": "Pending", "transactions": [{"status": "PENDING", "timestamp": 1700000000}]}, {"title": "This week", "transactions": [{"status": "CONFIRMED", "timestamp": 1697068759.077}]}, {"title": "2022", "transactions": [{"status": "CONFIRMED", "timestamp": 1650000000}]}]

      31 |     )
      32 |
    > 33 |     expect(result).toContainEqual(
         |                    ^
      34 |       expect.objectContaining({
      35 |         title: 'Today',
      36 |         transactions: expect.arrayContaining([

      at Object.<anonymous> (src/components/AccountDrawer/MiniPortfolio/Activity/utils.test.ts:33:20)
```

check the action report at the bottom of the pr

JFrankfurt commented:
@Jdecristi I'd try merging upstream's main into your branch to see if that fixes your CI problems.

Jdecristi commented:
@JFrankfurt I just ran a rebase on to updated main hope this helps.

Also sorry for the infrequent responses have little time now new job and learning solidity

JFrankfurt commented on src/pages/AddLiquidity/WrappedCurrencyToggle.tsx:
```diff
@@ -0,0 +1,65 @@
+  return (
+    <>
+      {native && wrappedNative ? (
+        <div style={{ width: 'fit-content', display: 'flex', alignItems: 'center' }} onClick={handleToggle}>
```
replace the inline style with:

```
const ToggleElementWrapper = styled.div`
  width: fit-content;
  display: flex;
  align-items: center;
`;
```

or better yet, find a way to wrap the existing `ToggleWrapper` component i.e., 
```
const ToggleElementWrapper = styled(ToggleWrapper)`
  width: fit-content;
  display: flex;
  align-items: center;
`;
```

so that you don't have a `div>div` situation here

JFrankfurt commented on src/pages/AddLiquidity/WrappedCurrencyToggle.tsx:
```diff
@@ -0,0 +1,65 @@
+  }
+
+  if (!currencyIsNative && !currencyIsWrappedNative) return null
+
```
```suggestion
  if (!currencyIsNative && !currencyIsWrappedNative) return null;
```

JFrankfurt commented on src/pages/AddLiquidity/WrappedCurrencyToggle.tsx:
```diff
@@ -0,0 +1,65 @@
+  }
+
+  if (!currencyIsNative && !currencyIsWrappedNative) return null
+
+  return (
+    <>
+      {native && wrappedNative ? (
```
double check that the native and wrappedNative checks here are necessary--it looks like the line above may already be handling this case? (Though maybe that should be `||` instead of `&&`?)

```suggestion
  if (!native || !wrappedNative) return null
  return (
```

this lets you substantially simplify the template

Jdecristi replied:
Hey @JFrankfurt,

isCurrencyNative & isCurrencyWrappedNative are just checks to make sure one of the currencies provided (currencyIdA or currencyIdB) are native or wrapped.

native and wrappedNative are the actual wrapped currencies to shown in the toggle the conditional is to make sure they are loaded in properly

I made is more clear in my most recent commit

JFrankfurt commented:
Thank god CI is finally passing. Sorry about that.

just-toby commented on src/pages/AddLiquidity/WrappedCurrencyToggle.tsx:
```diff
@@ -0,0 +1,61 @@
+  }
+
+  return (
+    <div style={{ width: 'fit-content', display: 'flex', alignItems: 'center' }} onClick={handleToggle}>
```
this should be a styled component too

Jdecristi replied:
Done

just-toby commented on src/pages/AddLiquidity/WrappedCurrencyToggle.tsx:
```diff
@@ -0,0 +1,61 @@
+
+  if (!native || !wrappedNative) return null
+
+  // checking to see if currencyIdA of currencyIdB is native or wrapped native
```
typo here

just-toby commented on src/pages/AddLiquidity/WrappedCurrencyToggle.tsx:
```diff
@@ -0,0 +1,61 @@
+type WrappedCurrencyToggleProps = {
+  currencyIdA?: string
+  currencyIdB?: string
+  onChangeCurrencies: (CurrencyAId?: string, currencyBId?: string) => void
```
make the capitalization match here plz

Jdecristi replied:
Done

just-toby reviewed (changes requested):
just a few nits, but you'll have to merge main anyway to fix the cloud tests (they should be fixed on main)
